proto-gen/api_v2/codec.go

```diff
@@ -0,0 +1,45 @@
+package api_v2
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc/encoding"
+	gproto "google.golang.org/protobuf/proto"
+	"reflect"
+	"strings"
+)
+
+func init() {
+	encoding.RegisterCodec(newCodec())
```

Comment:
@bogdandrutu or @tigrannajaryan could you please loop in here? This PR updates to the latest gRPC and protobuf. 

Jaeger uses gogo proto as OTELcol and it uses the same features - custom type for IDs and nonullable fields. To make this with the latest dependencies this codes registration is needed.  I could not find the custom codec registration in OTELcol. Do you know why it is not needed?

Reply:
>Do you know why it is not needed?

I am not very familiar with this specific functionality of gogo, I don't know the answer to this.